gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/BaseEntity.java

```diff
@@ -138,43 +139,120 @@ public B set(String name, String value) {
+      List<LongValue> values = new LinkedList<>();
+      values.add(of(first));
+      values.add(of(second));
+      for (long other : others) {
+        values.add(of(other));
+      }
+      properties.put(name, of(values));
+      return self();
+    }
+
     public B set(String name, double value) {
       properties.put(name, of(value));
       return self();
     }
 
+    public B set(String name, double first, double second, double... others) {
+      List<DoubleValue> values = new LinkedList<>();
+      values.add(of(first));
+      values.add(of(second));
+      for (double other : others) {
+        values.add(of(other));
+      }
+      properties.put(name, of(values));
+      return self();
+    }
+
     public B set(String name, boolean value) {
       properties.put(name, of(value));
       return self();
     }
 
+    public B set(String name, boolean first, boolean second, boolean... others) {
+      List<BooleanValue> values = new LinkedList<>();
+      values.add(of(first));
+      values.add(of(second));
+      for (boolean other : others) {
+        values.add(of(other));
+      }
+      properties.put(name, of(values));
+      return self();
+    }
+
     public B set(String name, DateTime value) {
       properties.put(name, of(value));
       return self();
     }
 
+    public B set(String name, DateTime first, DateTime second, DateTime... others) {
+      List<DateTimeValue> values = new LinkedList<>();
+      values.add(of(first));
+      values.add(of(second));
+      for (DateTime other : others) {
+        values.add(of(other));
+      }
+      properties.put(name, of(values));
+      return self();
+    }
+
     public B set(String name, Key value) {
       properties.put(name, of(value));
       return self();
     }
 
+    public B set(String name, Key first, Key second, Key... others) {
+      List<KeyValue> values = new LinkedList<>();
+      values.add(of(first));
+      values.add(of(second));
+      for (Key other : others) {
+        values.add(of(other));
+      }
+      properties.put(name, of(values));
+      return self();
+    }
+
     public B set(String name, FullEntity<?> value) {
       properties.put(name, of(value));
       return self();
     }
 
+    public B set(String name, FullEntity<?> first, FullEntity<?> second, FullEntity<?>... others) {
+      List<EntityValue> values = new LinkedList<>();
+      values.add(of(first));
+      values.add(of(second));
+      for (FullEntity<?> other : others) {
+        values.add(of(other));
+      }
+      properties.put(name, of(values));
+      return self();
+    }
+
     public B set(String name, List<? extends Value<?>> values) {
       properties.put(name, of(values));
       return self();
     }
 
-    public B set(String name, Value<?> value, Value<?>... other) {
-      properties.put(name, of(value, other));
+    public B set(String name, Value<?> first, Value<?> second, Value<?>... other) {
```

Comment:
Because of [this overloaded set](https://github.com/GoogleCloudPlatform/gcloud-java/blob/master/gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/BaseEntity.java#L131) method, I believe it's impossible to set a list with one value at the moment.